### PR TITLE
feat: Devcard Carousel

### LINKED ISSUE #1277: Devcard Screen: Create card carousel
[Figma](https://www.figma.com/file/hF1wdcQDIcgWtpbqRIEzNB/DevCards?type=design&node-id=584%3A1543&t=2XJamZVwrzNqcALV-1)

- Set of profiles is a pre-selected list of 10-20 people, we'll hardcode this initially
- Cards flip around in an infinite carousel
- Can navigate by:
  - clicking on the stack of cards under the first one
  - using the arrow keys, 
  - swipe left/right gesture on mobile
- Clicking the top card will cause it to flip around to see the back

### PR DESCRIPTION
**Pull request by foxyblocks**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
Implements DevCardCarousel component and adds it to the `/user/[profile]/card` route. Still only using stub data 

## Related Tickets & Documents
closes #1277 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/insights/assets/555044/c69ae71d-d202-42d4-9659-d0232df04201

https://github.com/open-sauced/insights/assets/555044/0641e64e-2a6e-4fcf-9b1a-5ad709826e47

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
QA'd and it works. I will let @OgDev-01 and @brandonroberts give the the code a review. I do appreciate you adding test for this. I will be curious how the page loads when the live api data is added. 

No need to implement, but it would be cool if the cards shuffled when you shake the phone.

File: pages/user/[username]/card.tsx [hunk 1]
```diff
@@ -20,22 +26,107 @@ export const getServerSideProps: GetServerSideProps<CardProps> = async (context)
...
+              <DevCardCarousel isLoading={false} cards={cardData} onSelect={(name) => setSelectedUserName(name)} />
+              <div className="hidden md:flex align-self-stretch justify-center">
+                <Button variant="primary" className="justify-center">
+                  Create your own dev card!
+                </Button>
+              </div>
+            </div>
+          </div>
+          <div className="hidden md:grid overflow-hidden">
+            {iframeTransition((style, item) => {
```
Comment (OgDev-01) [inline pages/user/[username]/card.tsx:77]: will changing `item`  to `username` for readability reasons break anything?
```suggestion
            {iframeTransition((style, username) => {
```

File: pages/user/[username]/card.tsx [hunk 2]
```diff
@@ -20,22 +26,107 @@ export const getServerSideProps: GetServerSideProps<CardProps> = async (context)
...
+                  className="grid overflow-hidden relative"
+                  style={{
+                    ...style,
+                    gridArea: "1 / 1",
+                    willChange: "transform, opacity",
+                  }}
+                >
+                  <div className="grid hover:scale-[1.01] cursor-pointer transition-all group">
+                    <div className="rounded-l-3xl h-full max-h-full grid bg-slate-50 overflow-hidden">
+                      {/* <div className="grid absolute place-content-center w-full h-full z-0 left-0 top-0">
```
Comment (OgDev-01) [inline pages/user/[username]/card.tsx:89]: nit: remove redundant codes

File: pages/user/[username]/card.tsx [hunk 3]
```diff
@@ -20,22 +26,107 @@ export const getServerSideProps: GetServerSideProps<CardProps> = async (context)
...
+const springFunction = (selectedIndex: number) => (index: number) => ({
+  opacity: index === selectedIndex ? 1 : 0,
+  transform: `translate3d(${index === selectedIndex ? 0 : 100}%, 0, 0)`,
+  config: {
+    easing: easings.easeInOutElastic,
+  },
+});
 
 const Card: NextPage<CardProps> = ({ username }) => {
+  const [selectedUserName, setSelectedUserName] = useState<string>(username as any);
```
Comment (OgDev-01) [inline pages/user/[username]/card.tsx:40]: why any?

**Review by OgDev-01 — CHANGES_REQUESTED** [review]
Hey @foxyblocks , will you move the `dev-card-carousel.stories.tsx` to the `stories/organisms` directory to keep codebase consistency?

File: components/molecules/DevCard/dev-card.tsx [hunk 4]
```diff
@@ -95,7 +103,7 @@ export default function DevCard(props: DevCardProps) {
   function handleCardClick(event: React.MouseEvent<HTMLElement, MouseEvent>) {
     // flip the card if the click is not on the button
     if (!(event.target instanceof HTMLAnchorElement || event.target instanceof HTMLButtonElement)) {
-      if (props["flipped"] === undefined) {
+      if (props["isFlipped"] === undefined) {
```
Comment (brandonroberts) [inline components/molecules/DevCard/dev-card.tsx:106]: Question: Why props["isFlipped"] here where other usage is `props.isFlipped`?

File: pages/user/[username]/card.tsx [hunk 5]
```diff
@@ -1,5 +1,11 @@
-import DevCard from "components/molecules/DevCard/dev-card";
+import Button from "components/atoms/Button/button";
+import HeaderLogo from "components/molecules/HeaderLogo/header-logo";
+import DevCardCarousel from "components/organisms/DevCardCarousel/dev-card-carousel";
+import BubbleBG from "../../../img/bubble-bg.svg";
+import { STUB_DEV_CARDS } from "components/organisms/DevCardCarousel/stubData";
 import { GetServerSideProps, NextPage } from "next";
+import { useEffect, useState } from "react";
+import { useTransition, animated, useSprings, easings } from "react-spring";
```
Comment (brandonroberts) [inline pages/user/[username]/card.tsx:8]: nit: keep third-party imports at the top (not a blocker)

```suggestion
import { GetServerSideProps, NextPage } from "next";
import { useEffect, useState } from "react";
import { useTransition, animated, useSprings, easings } from "react-spring";
```

**Review by brandonroberts — APPROVED** [review]
LGTM aside from review comments already left

**Comment by foxyblocks** [comment]
> No need to implement, but it would be cool if the cards shuffled when you shake the phone.

That would be cool, but there's no shake gesture in web js, there's an unmaintained library that does it https://github.com/alexgibson/shake.js/

How about we work on adding a drag/swipe gesture once the data piece is done? 😉

**Pull request merged**